### PR TITLE
feat(journey-maps): migrate journey-maps to sbb-angular

### PR DESCRIPTION
**Pull request by mstankala**

### CODE REVIEW

**Comment by mstankala** [comment]
@kyubisation the api_key is not available/configured in the staging showcaseApp. Can you support me?

**Review by kyubisation — CHANGES_REQUESTED** [review]
Some general feedback:
- We prefix our symbols (classes, constants, interfaces) with Sbb* or SBB_* and don't use the *Component, *Directive and *Service suffix. (e.g. `JourneyMapsClientComponent` => `SbbJourneyMapsClient`)
- The internal components still use the rokas selector prefix. Is this intended?
- Documentation is generally lacking. Is this really enough?

File: src/journey-maps/angular/angular.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Journey Maps for Angular
+
+`<sbb-journey-maps language="en" apiKey="<API-KEY>"></sbb-journey-maps>` displays an interactive map, with rail-specific style.
```
[the remark this answers is not in the document]
Comment (mstankala) [reply src/journey-maps/angular/angular.md]: The Documenation-Story is planed in next PI.

File: src/components-examples/journey-maps/angular/angular/angular-example.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  UIOptions,
+  ViewportOptions,
+  ZoomLevels,
+} from '@sbb-esta/journey-maps';
+import { FeatureCollection } from 'geojson';
+import { LngLatBoundsLike, LngLatLike } from 'maplibre-gl';
+import { BehaviorSubject, Subject, take } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { zhShWaldfriedhof } from './test-data/journey/zh-sh_waldfriedhof';
```
[the remark this answers is not in the document]
Comment (sebisteiner) [reply src/components-examples/journey-maps/angular/angular/angular-example.ts]: Yes. The data should be a blackbox for the regular user. He receives it from our backend (Journey Maps) and puts it directly into Journey Maps Client.

File: src/journey-maps/angular/components/leit-poi/model/leit-poi-placement.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export enum LeitPoiPlacement {
```
[the remark this answers is not in the document]
Comment (sebisteiner) [reply src/journey-maps/angular/components/leit-poi/model/leit-poi-placement.ts]: We kept `MarkerCategory` and `MarkerPriority` and replaced the rest with union types.

**Comment by sebisteiner** [comment]
> Documentation is generally lacking. Is this really enough?

I created some more documentation.

IMO we should split up the full example into multiple, smaller ones that showcase each one a different piece of functionality.

But for the moment I think we can start with what we have now.

File: src/journey-maps/angular/components/popup/popup.scss [hunk 4]
```diff
@@ -0,0 +1,2 @@
+// Popup style needs to be global
```
Comment (kyubisation) [inline src/journey-maps/angular/components/popup/popup.scss:1]: Nitpick: Why not simply declare global popup styles here and disable view encapsulation for this component?

**Review by kyubisation — CHANGES_REQUESTED** [review]
Some additional notes, but looks much closer to done.
Thanks for your work!

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Thank you for your adaptations!

**Pull request merged**